crates/matrix-sdk/src/latest_events/latest_event.rs

```diff
@@ -150,6 +169,332 @@ impl LatestEventValue {
             .map(Self::Remote)
             .unwrap_or_default()
     }
+
+    /// Create a new [`LatestEventValue::LocalIsSending`] or
+    /// [`LatestEventValue::LocalIsWedged`].
+    async fn new_local(
+        send_queue_update: &RoomSendQueueUpdate,
+        buffer_of_values_for_local_events: &mut LatestEventValuesForLocalEvents,
+        room_event_cache: &RoomEventCache,
+        power_levels: &Option<(&UserId, RoomPowerLevels)>,
+    ) -> Self {
+        use crate::send_queue::{LocalEcho, LocalEchoContent};
+
+        match send_queue_update {
+            // A new local event is being sent.
+            //
+            // Let's create the `LatestEventValue` and push it in the buffer of values.
+            RoomSendQueueUpdate::NewLocalEvent(LocalEcho {
+                transaction_id,
+                content: local_echo_content,
+            }) => match local_echo_content {
+                LocalEchoContent::Event { serialized_event: content, .. } => {
+                    if let Ok(content) = content.deserialize() {
```

Comment:
Can you `warn!` with the error, when deserialization failed?

Reply:
Yup, sure. I've done that for all `deserialize()` calls.